src/js/services.js

```diff
@@ -63,11 +63,24 @@ const getUserTimeline = (callback) => {
 		.then(checkStatus)
 		.then(parseJSON)
 		.then(function(data) {
-			callback(data, null);
+			callback(data, false);
+		})
+		.catch(function(error) {
+			callback(null, true);
+		})
+}
+
+const getFilterTimeline = (callback, filter) => {
```

Comment:
for promises, are you fully using promises properly? this comment applies to the getHomeTimeline as well